runtime/metricsview/ast.go

```diff
@@ -43,6 +43,9 @@ type SelectNode struct {
 	IsCTE                bool             // Whether this node is a Common Table Expression
 	DimFields            []FieldNode      // Dimensions fields to select
 	MeasureFields        []FieldNode      // Measure fields to select
+	IsInline             bool             // Whether this node is an inline view
+	InlineDimFields      [][]FieldNode    // Dimension fields to select in the inline view, row based representation of the dimensions. Each index in the outer array represents a row in the inline view.
+	InlineMeasureFields  [][]FieldNode    // Measure fields to select in the inline view, row based representation of the measures. Each index in the outer array represents a row in the inline view.
```

Comment:
Based on our sync discussion, would it make sense instead to:
1. Support a field like: `RawSelect *ExprNode // Raw SELECT statement with args. If set, other fields like FromSelect, Where, etc. are ignored.`
2. Populating `DimFields` and `MeasureFields` with the columns returned by the `RawSelect` subquery (instead of having separate `InlineXXX` variants)

(This is just a suggestion, not sure if I have enough context yet to determine if it's the best approach.)

Reply:
I have not tried `ExprNode` approach, just was sql as of now. Will try it out